sdk/eventhub/Azure.Messaging.EventHubs/src/Core/Guard.cs

```diff
@@ -92,17 +36,15 @@ internal static class Guard
         ///   throwing an <see cref="ArgumentOutOfRangeException" /> if that invariant is not met.
         /// </summary>
         ///
-        /// <param name="argumentName">The name of the argument being considered.</param>
         /// <param name="argumentValue">The value of the argument to verify.</param>
-        /// <param name="maxmimumLength">The maximum allowable length for the <paramref name="argumentValue"/>; its length must be less than or equal to this value.</param>
-        ///
-        public static void ArgumentNotTooLong(string argumentName,
-                                              string argumentValue,
-                                              int maxmimumLength)
+        /// <param name="maximumLength">The maximum allowable length for the <paramref name="argumentValue"/>; its length must be less than or equal to this value.</param>
+        /// <param name="argumentName">The name of the argument being considered.</param>
+        /// <exception cref="ArgumentOutOfRangeException"><paramref name="argumentValue"/> exceeds <paramref name="maximumLength"/> characters.</exception>
+        public static void NotTooLong(string argumentValue, int maximumLength, string argumentName)
         {
-            if (argumentValue?.Length > maxmimumLength)
+            if (argumentValue != null && argumentValue.Length > maximumLength)
             {
-                throw new ArgumentOutOfRangeException(String.Format(CultureInfo.CurrentCulture, Resources.ArgumentStringTooLong, argumentName, maxmimumLength), argumentName);
+                throw new ArgumentOutOfRangeException(argumentName, $"The argument '{argumentName}' cannot exceed {maximumLength} characters.");
```

Comment:
Is there a reason that these are being changed from the current resource set?

Reply:
There are a few reasons.

1. Since we're using shared source for the main `Argument` class from Azure.Core, sharing the resources to support it would be extra work and error prone. And since we aren't for that, it would be inconsistent for some developer exceptions to be localized and others not.
2. For *developer exceptions* it's often easier to find answers when the exception isn't localized, especially when those exception cases are documented. ;) Talked with @pakrym who brought this up with cases he's seen.
3. No other track 2 service is currently localizing developer exceptions.

Reply:
To be transparent, the only reason that the strings are resources was really inertia from track one.  They're not truly localized, only lightly applying culture formatting to dates and numeric values.   

I find value in having a centralized dictionary of messages just to keep verbiage consistent across the code base, but as `Argument` is already a central point where messages aren't re-used, I've no objections to keeping the changes.   I asked more from curiosity than any objection.